Submission2

```diff
@@ -0,0 +1,142 @@
+/* ----------------------------------------------------------------------------
+ * Copyright &copy; 2015 Jaemin Jin <jaemin_95@csu.fullerton.edu>
+ * Released under the [MIT License] (http://opensource.org/licenses/MIT)
+ * ------------------------------------------------------------------------- */
+
+/**
+ * A program to read unsigned integers from a text file and separate them into
+ * two files, one for primes, another for composites.
+ */
+
+#include <iostream>
+using std::cin;
+using std::cout;
+using std::endl;
+
+#include <string>
+using std::string;
+
+#include <fstream>
+using std::ifstream;
+using std::ofstream;
+
+int find_max(const string& infilename)              //DONE this part!
+{
+    int highestnumber, current;
+    std::ifstream infile;
+    infile.open("input.txt");
+    
+    infile >> current;
+    highestnumber = current;
+    
+    while (infile)
+    {
+        infile >> current;
+        if (current > highestnumber)
+            highestnumber = current;
+    }
+    
+    return highestnumber;
+}
+
+void sieve(const int size, bool primes[])          //DONE this part!
+{
+    for (int i = 2; i <= size; i++)
+    {
+        if (primes[i] == true)
+        
+            std::cout << i;
+        
+            for (int j = (i * i); j <= i; j++)
+            {
+                primes[j] = false;
+            }
+    }
+}
+
+int write_primes(const int size, bool primes[], const string& infilename, const string& outfilename) //Incomplete
+{
+    int p;
+    
+    std::ifstream infile("input.txt");
+    if (!infile.is_open())
+        return -1;
+    
+    std::ofstream outfile("primes.txt");
+    if (!outfile.is_open())
+        return -1;
+    
+    while (infile >> p)
+    {
+        infile >> p;
+        if (primes[p] == 1)
+        {
+            outfile << p << endl;
+        }
+    }
+    
+    return p;
+}
+
+int write_composites(const int size, bool primes[], const string& infilename, const string& outfilename) // Incomplete
+{
+    int compositesnumber;
+    
+    std::ifstream infile("input.txt");
+    if (!infile.is_open())
+        return -1;
+    
+    std::ofstream outfile("primes.txt");
+    if (!outfile.is_open())
+        return -1;
+    
+    sieve(size, primes);
```

Comment:
no need to call this here -- this function should be *almost* the same as `write_primes`